### ISSUE TITLE
Buffer tags

### ISSUE THREAD

**Issue by datanoise** [ibhagwan/fzf-lua#523]
Is there an equivalent way to run `:BTags` from `fzf.vim` where it runs `ctags` on the current buffer? 
`:FzfLua btags` requires a prebuilt tag file, which is not what I am after.

**Comment by ibhagwan** [comment]
All `tags` providers require a tags file, I’m not sure how fzf.vim’s `BTags` works but if it uses tags it probably automatically generates a temporary tags file each run, that’s pretty easy to do with a very simple customization.

**Comment by datanoise** [comment]
I am sure it reads directly from the standard output of `ctags` without creating any temporary tag files. I was hoping to get something like this done with fzf-lua.

**Comment by ibhagwan** [comment]
https://github.com/ibhagwan/fzf-lua/commit/1983b542e3673ceb13854faa46278877c251a1a6 - update to this commit and use `:FzfLua btags ctags_gen=true` or `:lua require'fzf-lua'.btags({ ctags_gen=true }))`

Use `setup` If you wish for the setting to be applied permanently:
```lua
require'fzf-lua'.setup({
  btags = { ctags_gen = true }
})
```

Few notes:
- The command that will be piped into fzf is `ctags -f /dev/stdout <file>`
- ctags binary can be modified with `btags({ ctags_bin = "..." })`
- `/dev/stdout` can be modified with `btags({ ctags_out = "..." })`

If you can think of a better way other than `-f /dev/stdout` let me know and I'll make the necessary adjustments.

**Comment by ibhagwan** [comment]
https://github.com/ibhagwan/fzf-lua/commit/5b681119dda2d697ce9e3a16c73b14e050bed0c8 - this commit enables auto generation of ctags as fallback when the tags file doesn't exist.

- If `tags` file exists, it will be used (has priority over auto generation), otherwise tags will be automatically generated using `ctags -f - <file>`
- If you have an existing `tags` file and still wish to use the dynamically generated tags you can do so by supplying the `ctags_autogen = true` option


TL;DR, if you don't have a `tags` file you can update to this commit and use `:FzfLua btags`, rest is taken care of automatically.

**Comment by datanoise** [comment]
Thank you

**Comment by ibhagwan** [comment]
> Thank you

Test it out and lmk if this needs more work or we can close this issue :-)

**Comment by ibhagwan** [comment]
Closing, feel free to reopen should you need more help.
